packages/library-base/src/commonMain/kotlin/io/realm/notifications/ObjectChange.kt

```diff
@@ -17,6 +17,23 @@
 package io.realm.notifications
 
 import io.realm.RealmObject
+import io.realm.query.RealmSingleQuery
+
+/**
+ * This sealed class describe the possible changes that can be observed on a [RealmSingleQuery] flow.
+ */
+sealed interface QueryObjectChange<O : RealmObject> {
```

Comment:
`QueryObjectChange` sounds a bit strange to me, but not sure that `ObjectQueryChange` is any better and prefixing `ObjectChange` with `Query<X>` might make sense 🤔